homeassistant/components/tami4/__init__.py

```diff
@@ -20,15 +20,17 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     try:
         api = await hass.async_add_executor_job(Tami4EdgeAPI, refresh_token)
-    except Exception as ex:
+    except APIExceptions.RefreshTokenExpiredException as ex:
+        raise ConfigEntryError("API Refresh token expired") from ex
```

Comment:
In a follow-up PR please implement the reauth flow and change

```suggestion
        raise ConfigEntryAuthFailed("API Refresh token expired") from ex
```

Reply:
Tests doesn't pass with ConfigEntryAuthFailed as the reauth isn't supported

Reply:
> In a follow-up PR please implement the reauth flow and change

I meant that you should implement reauth and the requested change in a follow-up PR after merging this one.
Please leave the conversation open until the follow-up PR is created.

Reply:
Won't the test failing prevent merging?